PseudoLocalizer.Core/POFileFormatException.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Text;
+using Karambolo.PO;
+
+namespace PseudoLocalizer.Core
+{
+    /// <summary>
+    /// The exception that is thrown when a PO file format error is detected.
+    /// </summary>
+    [Serializable]
+#pragma warning disable CS3003 // Type is not CLS-compliant
```

Comment:
I don't think this should be just being ignored - the exception should be CLS compliant, or use a built-in exception type (`InvalidOperationException` is my usual go-to for lack of something better when adding something new can be avoided).

Reply:
If we don't use a custom exception, we won't be able to pass the error details back to the global tool for reporting purposes (it's the inner object that isn't CLS compliant). Also, by using something like InvalidOperationException, we wouldn't be able to skip a specific badly-formed file, as it would be unsafe to infer that an IOE was raised by the POProcessor.

Which route would you prefer me to go down?

Reply:
I just realised I could perform the error text extraction in the POProcessor and pass back a descriptive string(s) instead of the raw object in the custom data - would you prefer that?

Reply:
I guess if the custom exception had an `ICollection<string>` for the messages extracted internally, then the tool could just catch and operate on those?